department-of-veterans-affairs/caseflow
# Fix email address bug when converting hearing to virtual

Resolves [CASEFLOW-2665](https://vajira.max.gov/browse/CASEFLOW-2665)

### Description
There were two things going on when converting a hearing to virtual: 

~~1. The frontend is only sending email addresses if they have been edited, but the backend was calling `create_or_update_email_recipients` no matter what, causing it to fail with an "email can't be blank" error when no email addresses were included with in the payload. To fix this, we now only call that method if the payload includes email attributes.~~
1. The backend was missing an `if appellant_email.present?` check. But also, the frontend was not displaying errors returned by the backend because it expected VirtualHearingModal to display them, but that component no longer exists. Removed that logic, so frontend now displays the errors inline. 

2. Pre-submission "Email required" errors were only getting displayed for hearings that were already virtual, not those in the process of being converted to virtual. To fix this, I expanded the check to `hearing.isVirtual || convertingToVirtual`.

### More updates:

~~Needed to add safe navigation operators in `virtual_hearing.rb` due to the fact that requests now only include email addresses if they have been edited. For the same reason, updated a test in `hearing_controller_spec.rb` to expect a 200 response when the request does not include an appellant email.~~ 
Removed the outdated test in `hearing_controller_spec.rb`.

Beyond those updates, the rest of the changes are due to renaming `EmailModalConfirmation` to `EmailConfirmationModal` and renaming all methods that relate to that modal, so that they no longer refer to the now non-existent component `VirtualHearingModal`. 

### Testing Plan
Log in as BVASYELLOW

1. From http://localhost:3000/hearings/schedule, select a future day with non-virtual hearings, and find a Legacy hearing (note: I tried with both AMA and Legacy hearings, but the original issue specifies Legacy hearings.)
2. Click the "Edit Hearing Details" link for the legacy non-virtual hearing
3. Choose "Virtual" from the "Hearing Type" dropdown
4. Scroll to the bottom of the "Convert to Virtual" page that results, and click the blue "Convert to Virtual" button.
5. Verify that it succeeds.
6. Repeat steps 1 - 3, and then delete the veteran email. Click "Convert to Virtual" and verify that the page scrolls up to show you an error on the empty email address box.
